src/battle_util.c

```diff
@@ -6458,33 +6458,36 @@ static inline bool32 CanBreakThroughAbility(u32 battlerAtk, u32 battlerDef, u32
 
 u32 GetBattlerAbility(u32 battler)
 {
-    bool32 noAbilityShield = GetBattlerHoldEffectIgnoreAbility(battler, TRUE) != HOLD_EFFECT_ABILITY_SHIELD;
-    bool32 abilityCantBeSuppressed = gAbilitiesInfo[gBattleMons[battler].ability].cantBeSuppressed;
-
-    if (abilityCantBeSuppressed)
+    if (!gBattleStruct->aiCalcInProgress)
```

Comment:
The function now skips important checks for the AI like neutralizing gas but there is also an equivalent function `AI_GetBattlerAbility` that omits some of the redundant checks. We can combine them but the AI should still have the correct checks.

Reply:
Generally I think in this instance it makes more sense to use `AI_GetBattlerAbility` because it has only the essential checks needed and by combining them the code would get a bit ugly